torch/testing/_internal/common_jit.py

```diff
@@ -143,18 +143,78 @@ def getExportImportCopy(self, m, also_test_file=True, map_location=None):
             torch.jit.save(imported, fname)
             return torch.jit.load(fname, map_location=map_location)
 
+    def autoDiffErrorMessage(self, should_autodiff_node, nodes_not_in_diff_graph, 
+                             fusion_nodes_not_found, non_fusible_nodes_being_fused, 
+                             fusion_nodes_found, nodes_in_diff_graph):
+        err_msg = "\nFailure in testing nodes' autodifferentiation, "
+        if should_autodiff_node:
+            err_msg += "one or more nodes were expected to be autodiffed, " \
+                "but were not found in specified fusible/nonfusible " \
+                "DifferentiableGraph groups. \nSpecifically:"
+            for node in nodes_not_in_diff_graph:
+                err_msg += f"\n  {node} was not in one of the DifferentiableGraphs " \
+                    "when it was expected to be. Did you intend for this node to be " \
+                    "autodiffed? If not, remove it from the list of nonfusible nodes."
+                if node in non_fusible_nodes_being_fused:
```

Comment:
If `should_autodiff_node` is True, a `foo` node is in `nodes_not_in_diff_graph` and that node is also in `non_fusible_nodes_being_fused`, then I think the printed error message will be:

> Failure in testing nodes' autodifferentiation, one or more nodes were expected to be autodiffed but were not found in specified fusible/nonfusible DifferentiableGraph groups. 
> 
> Specifically: 
> `foo` was not in one of the DifferentiableGraphs when it was expected to be. Did you intend for this node to be autodiffed? If not, remove it from the list of nonfusible nodes. Additionally, This [sic] node was found in a FusionGroup in a DifferentiableGraph. If that was intended, reclassify this node as a fusible node. If not, your autodifferentiation logic might be wrong.

To someone not familiar with the JIT, it seems like the above is saying that `foo` is both in and not in a differentiable graph. Since it's in a fusion group in a differentiable graph but the paragraph for `foo` starts by saying it's not in a differentiable graph. 

I think the cases for `should_autodiff_node=True` are:

- [Case 1] the node is intended to appear in a differentiable graph but doesn't
- [Case 2] the node is intended to appear in a differentiable graph outside of a fusion group, but it's in a fusion group
- [Case 3] the node is intended to appear in a differentiable graph inside a fusion group, but it's not inside a fusion group

Is that correct? If so, I would suggest identifying which case the op falls in and then crafting a simplified error message for each case. For example,

[Case 1] "Node `foo` is marked as being autodifferentiable, but doesn't appear in any differentiable graph. If this is expected, ..."
[Case 2] "Node `foo` is marked as being nonfusible, but appears in a fusion group. If `foo` is expected to be fused then remove it from the list of nonfusible nodes." 
[Case 3] "Node `foo` is marked as being fusible, but appears outside a fusion group. If `foo` is expected to be unfused then add it to the list of nonfusible nodes."

Does that makes sense or am I misunderstanding these cases?

Unfortunately this proposal is much easier to implement per-op, which makes implementing a suggestion like @ellison's above a little trickier. The nodes could be added to three lists, one for each category, before the error messages are constructed, however.